src/models/Player.test.ts

```diff
@@ -0,0 +1,117 @@
+import { Game } from './Game'
+import { Dealer } from './Dealer'
+import { Player } from './Player'
+
+describe('player test', () => {
+  let game: Game
+  let dealer: Dealer
+  let playerA: Player
+  let playerB: Player
+  let playerC: Player
+  let players: Player[]
+
+  beforeEach(() => {
+    /**
+     *  initial setting up
+     */
+    game = new Game()
+    dealer = new Dealer()
+    playerA = new Player({ nickname: 'well-balanced', socketId: '1' })
+    game.joinPlayer(playerA)
+    playerB = new Player({ nickname: 'roseline', socketId: '2' })
+    game.joinPlayer(playerB)
+    playerC = new Player({ nickname: 'ohdoski', socketId: '3' })
+    game.joinPlayer(playerC)
+    players = [playerA, playerB, playerC]
+  })
+
+  it('sets deck into each player', () => {
+    const deck = dealer.createDeck()
+    const cardBundle = dealer.dealCards(deck, 3)
+    players.map((player) => {
+      player.setDeck(cardBundle.pop())
+      expect(player.deck).toBeTruthy()
+    })
+  })
+
+  it('sets pass state to true', () => {
+    game.activePlayer = playerA
+    /**
+     * can not pass if player doesn't have deckSize
+     */
+    playerA.setDeck([1, 2])
+    playerA.passOn()
+    expect(playerA.pass).toBeTruthy()
+  })
+
+  test('the number of cards decreases if player play cards', () => {
+    const player = playerA
+    player.isFirst = true
+    const mockDeck = [11, 12, 12, 12, 12, 12]
+    player.setDeck(mockDeck)
+    game.activePlayer = player
+    player.playCards({ cardNumber: 12, numberOfCards: 5 })
+
+    expect(player.deck.length).toBe(1)
+  })
+
+  it(`throws an error if second player play card number
+  equal to or greather than card first player played`, () => {
+    const firstPlayer = playerA
+    game.activePlayer = firstPlayer
+    firstPlayer.isFirst = true
+    firstPlayer.setDeck([11, 11, 11])
+    firstPlayer.playCards({ cardNumber: 11, numberOfCards: 3 })
+
+    /**
+     * check current rule
+     */
+    expect(game.currentRule).toEqual({
+      currentCard: 11,
+      currentNumberOfCard: 3,
+    })
+
+    /**
+     * throw error if player play card equal to or greater than first number
+     */
+    const secondPlayer = game.activePlayer
+    secondPlayer.setDeck([11, 11, 11, 12, 12, 12])
+    expect(() => secondPlayer.playCards({ cardNumber: 11 })).toThrow()
+
+    /**
+     * set activePlayer to secondPlayer again
+     */
+    game.activePlayer = secondPlayer
+    expect(() => secondPlayer.playCards({ cardNumber: 12 })).toThrow()
+  })
+
+  test('player state(isFirst, pass) is initialized', () => {
+    const player = playerA
+    player.isFirst = true
+    player.pass = true
+    player.setDeck([1])
+    player.initShallowState()
+
+    expect(player.deck).toBeTruthy()
+    expect(player.deckSize).toBeTruthy()
+    expect(player.isFirst).toBeFalsy()
+    expect(player.pass).toBeFalsy()
```

Comment:
falsy에는 nullish한 것도 포함돼요

```suggestion
    expect(player.isFirst).toBe(false)
    expect(player.pass).toBe(false)
```

Reply:
감사합니다!